File: backend/src/main/scala/microservices/traveler/objects/TravelerBasicInfo.scala
// 本文件定义 traveler 模块的基础传输对象 `TravelerBasicInfo`，仅负责数据结构与 JSON codec。
package com.typesafe.travel.traveler.domain

import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}

final case class TravelerBasicInfo(
    fullName: String,
    gender: String,
    birthDate: String,
    nationality: String
)

object TravelerBasicInfo:
  given sourceEncoder: Encoder[TravelerBasicInfo] = deriveEncoder
  given sourceDecoder: Decoder[TravelerBasicInfo] = deriveDecoder

File: frontend/src/microservices/traveler/objects/TravelerBasicInfo.ts
// 本文件定义 traveler 模块的 `TravelerBasicInfo`，作为基础传输数据并提供 JSON 编解码。

export type TravelerBasicInfo = {
  fullName: string
  gender: string
  birthDate: string
  nationality: string
}

export const travelerBasicInfoFromJson = (json: string): TravelerBasicInfo =>
  JSON.parse(json) as TravelerBasicInfo

export const travelerBasicInfoToJson = (value: TravelerBasicInfo): string =>
  JSON.stringify(value)
